lib/vaahextendflutter/widgets/atoms/buttons.dart

```diff
@@ -341,10 +395,12 @@ class ButtonTextWithIcon extends StatelessWidget {
           fontSize: fontSize,
         ),
       ),
-      icon: FaIcon(
-        iconData,
-        size: iconSize,
-      ),
+      icon: assetPath != null
+          ? Image.asset(assetPath!)
```

Comment:
what if the asset is svg? also what if we separate vaahflutter to package, we wouldn't be able to use assets dir there, so instead pass widget and use that widget as leading.